crates/noirc_evaluator/src/ssa.rs

```diff
@@ -124,10 +111,40 @@ pub fn create_circuit(
     Ok((circuit, debug_info, abi))
 }
 
-impl Ssa {
-    fn print(self, print_ssa_passes: bool, msg: &str) -> Ssa {
-        if print_ssa_passes {
-            println!("{msg}\n{self}");
+// This is just a convenience object to bundle the ssa with `print_ssa_passes` for debug printing.
+struct SsaBuilder {
```

Comment:
nit: Just cause of the naming would it make sense to have this be a part of the `ssa_builder` module

Reply:
I wanted to keep this one local since it is just for debug printing so I renamed `ssa_builder` to `function_builder` instead since it only contains the FunctionBuilder.